src/projection_client.rs

```diff
@@ -0,0 +1,629 @@
+use crate::event_store::client::projections;
+use crate::event_store::client::shared::Empty;
+use crate::grpc::{ClientSettings, GrpcClient};
+use crate::Credentials;
+use futures::stream::BoxStream;
+use serde::de::DeserializeOwned;
+
+#[derive(Clone, Debug)]
+pub enum StatsFor {
+    Name(String),
+    AllProjections,
+    AllTransients,
+    AllContinuous,
+    AllOneTimes,
+}
+
+#[derive(Clone, Debug)]
+pub struct StatsDetails {
+    pub core_processing_time: i64,
+    pub version: i64,
+    pub epoch: i64,
+    pub effective_name: String,
+    pub writes_in_progress: i32,
+    pub reads_in_progress: i32,
+    pub partitions_cached: i32,
+    pub status: String,
+    pub state_reason: String,
+    pub name: String,
+    pub mode: String,
+    pub position: String,
+    pub progress: f32,
+    pub last_checkpoint: String,
+    pub events_processed_after_restart: i64,
+    pub checkpoint_status: String,
+    pub buffered_events: i64,
+    pub write_pending_events_before_checkpoint: i32,
+    pub write_pending_events_after_checkpoint: i32,
+}
+
+#[derive(Clone)]
+pub struct ProjectionClient {
+    client: GrpcClient,
+    settings: ClientSettings,
+}
+
+impl ProjectionClient {
+    pub async fn create(settings: ClientSettings) -> Result<Self, Box<dyn std::error::Error>> {
+        let client = GrpcClient::create(settings.clone()).await?;
+
+        Ok(ProjectionClient { client, settings })
+    }
+
+    pub async fn create_one_time_projection(
```

Comment:
**SUGGESTION:** I think that the `_projection` suffix is redundant. The user will already know that's creating projection by using `ProjectionClient` what will be created. Also, .NET gRPC client does not use this suffix.

The same comment applies for other methods (e.g. `create_transient_projection`, `update_projection`, etc.).

Reply:
Agree